addons/mail/models/res_users.py

```diff
@@ -43,13 +43,26 @@ def create(self, vals_list):
                 raise exceptions.RedirectWarning(msg, action.id, _('Go to the configuration panel'))
 
         users = super(Users, self).create(vals_list)
+        for user in users:
+            if user.has_group('base.group_portal'):
+                user.partner_id.message_post(body=user._get_portal_access_body(True), message_type='comment', subtype_xmlid='mail.mt_note')
         # Auto-subscribe to channels unless skip explicitly requested
         if not self.env.context.get('mail_channel_nosubscribe'):
             self.env['mail.channel'].search([('group_ids', 'in', users.groups_id.ids)])._subscribe_users_automatically()
         return users
 
     def write(self, vals):
+        user_portal_access_dict = {
+            user.id: user.has_group('base.group_portal')
+            for user in self
+        }
         write_res = super(Users, self).write(vals)
+        for user in self:
+            user_has_group = user.has_group('base.group_portal')
+            if user_has_group and not user_portal_access_dict[user.id]:
+                user.partner_id.message_post(body=user._get_portal_access_body(user_has_group), message_type='comment', subtype_xmlid='mail.mt_note')
+            elif not user_has_group and user_portal_access_dict[user.id]:
+                user.partner_id.message_post(body=user._get_portal_access_body(user_has_group), message_type='comment', subtype_xmlid='mail.mt_note')
```

Comment:
As you added a new method to build the message, this can be improved to avoid redundancy
```python
portal_access_changed = user_has_group != user_portal_access_dict[user.id]
if portal_access_changed:
    body = user._get_portal_access_body(user_has_group)
    user.partner_id.message_post(body=body, message_type='comment', subtype_xmlid='mail.mt_note')

```

Reply:
Changes done.